app/src/main/java/ch/epfl/sdp/blindly/weather/WeatherActivity.kt

```diff
@@ -163,4 +190,82 @@ class WeatherActivity : AppCompatActivity(), WeatherService.WeatherResultCallbac
     override fun onRefresh() {
         weather.nextWeek(location, callback = this)
     }
-}
\ No newline at end of file
+
+    /**
+     * Forms a new Calendar Event in Google Calendar App,
+     * closes the current activity as this page is no longer needed.
+     *
+     * @param title Title of the event to add to Calendar
+     * @param date Date of the event to add to Calendar
+     */
+    private fun setCalendarEvent(title: String, date: Calendar) {
+        val calIntent = Intent(Intent.ACTION_INSERT)
+        calIntent.data = Events.CONTENT_URI
+        calIntent.putExtra(Events.TITLE, title)
+        calIntent.putExtra(CalendarContract.EXTRA_EVENT_ALL_DAY, true)
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_BEGIN_TIME,
+            date.timeInMillis
+        )
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_END_TIME,
+            date.timeInMillis
+        )
+        finish()
+        startActivity(calIntent)
+    }
+
+    /**
+     * Takes the new date, compares with today and gets the result as an index.
+     * Then sends the result to setColoredBackgroundForSelectedDay to highlight
+     * the weather of the chosen day
+     *
+     * @param year Year of selected date
+     * @param month Month of selected date
+     * @param dayOfMonth Day of selected date
+     */
+    private fun computeDayIndexAndSetBackground(year: Int, month: Int, dayOfMonth: Int) {
+        calendar.set(year, month, dayOfMonth)
+        calendarView.date = calendar.timeInMillis
+        val selectedDate = Calendar.getInstance()
+        selectedDate.set(year, month, dayOfMonth)
+        val diff: Long = selectedDate.timeInMillis - today.timeInMillis
+        val dayIndex = TimeUnit.MILLISECONDS.toDays(diff)
+
+        setColoredBackgroundForSelectedDay(dayIndex)
+    }
+
+    /**
+     * If index is between the days we showed, (0 to 5 included, with 0 as today)
+     * then highlight that day, otherwise set the background to white.
+     *
+     * @param indexOfDay Index of the day chosen (with 0 as today)
+     */
+    private fun setColoredBackgroundForSelectedDay(indexOfDay: Long) {
+        for (i in 0..5) {
+            val layoutId = resources.getIdentifier(
+                "${WEATHER_PREFIX}${(i + 1)}",
+                "id",
+                packageName
+            )
+            if (i.toLong() == indexOfDay) {
+
+                findViewById<LinearLayout>(layoutId).setBackgroundColor(
+                    ContextCompat.getColor(
+                        applicationContext,
+                        R.color.blindly_blue_transparent
+                    )
+                )
+            } else {
+                findViewById<LinearLayout>(layoutId).setBackgroundColor(
+                    ContextCompat.getColor(
+                        applicationContext,
+                        R.color.white
```

Comment:
To get rid of the "Similar blocks of code" warning, maybe you could save the color with a ternary operation and then make only one call to setBackgroundColor with the saved color... But that's not really an issue to me